hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -847,14 +853,9 @@ object PruneDeadFields {
       case MatrixMapEntries(child, newEntries) =>
         val child2 = rebuild(child, memo)
         MatrixMapEntries(child2, rebuild(newEntries, child2.typ, memo))
-      case MatrixMapRows(child, newRow, newKey) =>
+      case MatrixMapRows(child, newRow) =>
```

Comment:
Do you need a case here for `MatrixKeyRowsBy`?

Reply:
nope, the default case for rebuild is to rebuild the children and then rebuild me. The reason that the things with value IRs (like MatrixMapRows) need to have special rebuild rules is that they need the rebuilt child type to rebuild the value IRs